extensions-core/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentMover.java

```diff
@@ -155,40 +158,39 @@ private void selfCheckingMove(
       String s3Path,
       String targetS3Path,
       String copyMsg
-  ) throws ServiceException, IOException, SegmentLoadingException
+  ) throws IOException, SegmentLoadingException
   {
     if (s3Bucket.equals(targetS3Bucket) && s3Path.equals(targetS3Path)) {
       log.info("No need to move file[s3://%s/%s] onto itself", s3Bucket, s3Path);
       return;
     }
-    if (s3Client.isObjectInBucket(s3Bucket, s3Path)) {
-      final S3Object[] list = s3Client.listObjects(s3Bucket, s3Path, "");
-      if (list.length == 0) {
+    if (s3Client.doesObjectExist(s3Bucket, s3Path)) {
+      final ListObjectsV2Result listResult = s3Client.listObjectsV2(
+          new ListObjectsV2Request()
+              .withBucketName(s3Bucket)
+              .withPrefix(s3Path)
+              .withMaxKeys(1)
+      );
+      if (listResult.getKeyCount() == 0) {
         // should never happen
         throw new ISE("Unable to list object [s3://%s/%s]", s3Bucket, s3Path);
       }
-      final S3Object s3Object = list[0];
-      if (s3Object.getStorageClass() != null &&
-          s3Object.getStorageClass().equals(S3Object.STORAGE_CLASS_GLACIER)) {
-        throw new ServiceException(StringUtils.format(
+      final S3ObjectSummary objectSummary = listResult.getObjectSummaries().get(0);
+      if (objectSummary.getStorageClass() != null &&
+          StorageClass.fromValue(StringUtils.toUpperCase(objectSummary.getStorageClass())).equals(StorageClass.Glacier)) {
+        throw new ISE(
```

Comment:
This is to keep it from being caught in a retry loop? can a comment be added to the code here as such?

Reply:
Looks like my fault. I don't want to change any logic in this PR. Changed to AmazonServiceException.